python/util.py

```diff
@@ -22,6 +22,97 @@
 import array
 import struct
 from uuid import getnode as get_mac
+import winreg as wreg
+import json
+import urllib.request
+import urllib.parse
+
+# Utils for sample.py
+
+def get_scheduled_events(address, headers):
+    '''
+    Make a GET request and return the response received.
+    '''
+    request = urllib.request.Request(address, headers=headers)
+    return urllib.request.urlopen(request)
+
+def post_scheduled_events(address, data, headers):
+    '''
+    Make a POST request.
+    '''
+    request = urllib.request.Request(address, data=str.encode(data), headers=headers)
+    urllib.request.urlopen(request)
+
+def make_address(ip_address):
+    '''
+    Returns the address for scheduled event endpoint from the IP address provided.
+    '''
+    return 'http://' + ip_address + '/metadata/scheduledevents?api-version=2017-03-01'
+
+def check_ip_address(address, headers):
+    '''
+    Checks whether the address of the scheduled event endpoint is valid.
+    '''
+    try:
+        response = get_scheduled_events(address, headers)
+        return 'Events' in json.loads(response.read())
+    except (urllib.error.URLError, UnicodeDecodeError, json.decoder.JSONDecodeError) as _:
+        return False
+
+def get_ip_address_reg_env(args):
+    '''
+    Get the IP address of scheduled event from registry or environment.
+    Returns None if IP address is not provided or stored.
+    '''
+    ip_address = None
+    if args.use_registry and sys.platform == 'win32':
+        # use CloudControlIp in registry if available.
+        try:
+            key = wreg.OpenKey(wreg.HKEY_LOCAL_MACHINE, "Software\\CloudControl")
```

Comment:
recommend moving the import winreg to down here after the sys.platform check, otherwise this will never run on Linux (or do something like this https://stackoverflow.com/questions/29289411/import-winreg-while-retaining-linux-support)